crates/wasi-provider/src/states/registered.rs

```diff
@@ -0,0 +1,156 @@
+use log::{error, info};
+
+use crate::{make_status, PodState};
+use kubelet::state::{State, Transition};
+use kubelet::{
+    container::Container,
+    pod::{Phase, Pod},
+    state,
+};
+
+use super::error::Error;
+use super::image_pull::ImagePull;
+
+fn validate_pod_runnable(pod: &Pod) -> anyhow::Result<()> {
+    if !pod.init_containers().is_empty() {
+        return Err(anyhow::anyhow!(
+            "Cannot run {}: spec specifies init containers which are not supported on wasCC",
+            pod.name()
+        ));
+    }
+    for container in pod.containers() {
+        validate_container_runnable(&container)?;
+    }
+    Ok(())
+}
+
+fn validate_container_runnable(container: &Container) -> anyhow::Result<()> {
+    if has_args(container) {
+        return Err(anyhow::anyhow!(
+            "Cannot run {}: spec specifies container args which are not supported on wasCC",
```

Comment:
TODO: validate whether the WASI provider accepted container arguments.

Reply:
It did.